src/modules/communication/internals/sendgrid-email-params.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  LooseObject,
+  SendEmailParams,
+  SendGridValidationError,
+  ValidationFailure
+} from "../types";
+
+export class SendGridEmailParams {
```

Comment:
This is not specific to send grid I believe, best to rename it to SendEmailPArams